### DISCUSSION TITLE
can't update records or collections - deployed to fly.io

### DISCUSSION THREAD

**Discussion by burzumumbra** [pocketbase/pocketbase#750] [Q&A] [answered]
Pocketbase is deployed in Fly.io
all was ok, I was setting up S3 and email, then tried to modify an image that was added before the S3 config, but I would return 502, also editing a collection fails. 

Already tried restarting the app in fly.io, but without luck. 

![image](https://user-images.githubusercontent.com/4171009/194771478-21966706-69ad-44ff-afa9-05eed2c63b2f.png)

**Comment by ganigeorgiev** [comment, marked as answer]
When switching from local <-> s3 files storage is up to the developers to migrate their old data. There should have been a warning that looks something like this:

![storage_warning](https://i.imgur.com/YQWHLg2.png)

**Reply by burzumumbra** [reply]
But that would be related to not being able to update anything? Bc even other collections or records can't be updated.

**Reply by ganigeorgiev** [reply]
@burzumumbra But we are ignoring the delete file errors (you can check https://github.com/pocketbase/pocketbase/blob/master/forms/record_upsert.go#L392-L396).

You should be able to update the record and remove the old file (as long as you have a valid storage setup).

**Reply by ganigeorgiev** [reply]
The 502 error is not from PocketBase. If you are still having the same error, could you post here the error request from "Inspect tool > Network" tab?

**Reply by burzumumbra** [reply]
I couldn't get the 502 error again, but weird stuff that this other record in a different collection allowed me to upload once the S3 was set up, but now returns a 400 error trying to update the image or other fields.
![image](https://user-images.githubusercontent.com/4171009/195000336-1641afc5-c37e-46bc-8ccb-801c06dc66aa.png)

**Reply by ganigeorgiev** [reply]
@burzumumbra Both fields from the screenshot (shelter and profile_photo) seems to be marked as "Required" so the validation is correct in this case.

**Reply by burzumumbra** [reply]
Sorry, forgot to send another screen, where I cleared another field that wasn't required, and returned 400 too since I can't figure this out. I'll just take note of first doing all the config for the project, S3, email, etc. then start working on collections. Thanks for the help and the fast replay loved that support.
